### DISCUSSION TITLE
Plotting like inverse pole figure

### DISCUSSION THREAD

**Discussion by mfuhr-mpg** [mtex-toolbox/mtex#1376] [Ask Anything] [answered]
Hi everyone, 

first of all (since this can't be emphasized strongly enough): Thank you very much for the great toolbox, I really appreciate your dedication and the great support!

Now to my question: I need a function that does more or less the same as the algorithm that is used for inverse pole figure plots , i.e. I have an orientation (in my case in Euler notation) as an input variable and would like to get the corresponding colour from the IPF colour key as an output. Is there already an appropriate function somewhere in the depths of MTEX that I can use or does anyone have an idea how to implement my idea?

Thanks in advance!

Best
Max

**Comment by kilir** [comment, marked as answer]
Hi Max,
some ways to do that:

```
cs = crystalSymmetry('321')
o = orientation.rand(100,cs)
% some reference direction
refdir = vector3d(2,1,0)

% option 1
ck= HSVDirectionKey(cs)
colors = ck.direction2color(inv(o).*refdir)

% option 2
ck = ipfHSVKey(cs)
ck.inversePoleFigureDirection = refdir
colors = ck.orientation2color(o)
```
Hope that helps.
Cheers,
Rüdiger

**Reply by mfuhr-mpg** [reply]
Quick and helpful as always - thanks, that works like a charm!
